### PR TITLE
dialects: Add PrintfOp to snrt dialect

### PR DESCRIPTION
**Pull request by JosseVanDelm**
This PR adds a printfop to the snrt dialect as per @nazavode 's request.
https://github.com/pulp-platform/snitch/blob/b9fe5550e26ea878fb734cfc37d161f564252305/sw/snRuntime/vendor/printf.c

Right now it is pretty low level though (reads in an 8bit/char LLVM pointer to the format string) which is a bit unfortunate:
Because right now it will be lowered to something like this:
https://github.com/xdslproject/xdsl/wiki/IO-Example-through-LLVM-stdlib

At some point (when directly lowering to RISC-V assembly, we might want to change this though, but lowering printf to RISC-V seems a bit non-trivial, and maybe not something we want to go for.
The snitch runtime basically steals this implementation of printf from somewhere else anyway.

### CODE REVIEW

File: xdsl/dialects/snitch_runtime.py [hunk 1]
```diff
@@ -1,11 +1,21 @@
...
 from xdsl.ir import OpResult, Dialect
-from xdsl.dialects.builtin import IntegerType, Signedness, IndexType
+from xdsl.dialects.builtin import IntegerType, Signedness, IndexType, i32
+from xdsl.dialects.llvm import LLVMPointerType
 from typing import Annotated
 
 u32 = IntegerType(data=32, signedness=Signedness.UNSIGNED)
 u64 = IntegerType(data=64, signedness=Signedness.UNSIGNED)
+i8 = IntegerType(data=8, signedness=Signedness.SIGNLESS)
+charptr = LLVMPointerType.typed(i8)
```
Comment (superlopuh) [inline xdsl/dialects/snitch_runtime.py:18]: I'm not sure that we want a dependency on llvm for the snitch dialect. Is there any other way to represent this?

Comment (JosseVanDelm) [reply xdsl/dialects/snitch_runtime.py:18]: I completely agree, open to suggestions.
I could for example, let this be emitted as part of the lowering process to the LLVM specific part?

Right now it is implemented like this to make the linking possible as per this
wiki page:
https://github.com/xdslproject/xdsl/wiki/IO-Example-through-LLVM-stdlib

Comment (JosseVanDelm) [reply xdsl/dialects/snitch_runtime.py:18]: @AntonLydike do you have any suggestions?

Comment (webmiche) [reply xdsl/dialects/snitch_runtime.py:18]: Does anything prevent you from just adding a `SnrtString` type for the moment? Once you get a better handle on what that thing really is, you can still refactor, IMO. Or do you actually go through LLVM IR at some point to lower/execute this?

Comment (webmiche) [reply xdsl/dialects/snitch_runtime.py:18]: During lowering, you can then lower that to the equivalent of a char* in whatever comes after snrt, i guess.

Comment (tobiasgrosser) [reply xdsl/dialects/snitch_runtime.py:18]: I am actually curious what you want to print? Printf is a C style interface with complex formatting. Do we need such an interface, or would it be sufficient to have something simpler, e.g., just print_int()?

**Pull request closed without merging**